### PR TITLE
Applying downward pressure on ratings.

### PR DESCRIPTION
**Pull request by 48klocs**
tl;dr - a repeated complaint is that everything is rated too highly, so I'm trying to achieve some separation in ratings.

We generally don't get a lot of downvotes on items, but we (ironically?) do get more of those on items popular in the meta.

I made the algo a little more sensitive to downvotes - we weight them more heavily, but the weighting we applies disappates as the number of total votes grows. Items people don't care much about that have a smattering of downvotes will look worse, items with oodles of reviews (good and bad) we'll report what the numbers give us.

A few before/after numbers (for PC platform ratings):

- Nameless Midnight
  - 4.9 / 4.9
- Better Devils
  - 4.9 / 4.9
- Hard Light
  - 4.7 / 4.5
- Uriel's Gift
  - 4.9 / 4.8
- Scathelocke
  - 4.9 / 4.9
- Lincoln Green
  - 4.6 / 4.1
- Skyburner's Oath
  - 3.7 / 1.8
- Sunshot
  - 5.0 / 5.0
- MIDA Multi-Tool
  - 4.2 / 4.2
- MIDA Mini-Tool
  - 4.6 / 3.9
- Inaugural Address
  - 4.3 / 3.3
- Tractor Cannon
  - 3.6 / 1.6

### CODE REVIEW

File: src/app/destinyTrackerApi/d2-reviewDataCache.js [hunk 1]
```diff
@@ -44,11 +44,29 @@ class D2ReviewDataCache {
     return rating.toFixed(1);
   }
 
+  _getDownvoteMultiplier(dtrRating) {
```
Comment (kyleshay) [inline src/app/destinyTrackerApi/d2-reviewDataCache.js]: love this idea. how could it scale? over time everything may have over 300 ratings. similarly if we reset scores per-season everything will have a -2.5 multiplier. 

could we build a scale based off of the average number of all reviews or something? 🤔

Comment (kyleshay) [reply src/app/destinyTrackerApi/d2-reviewDataCache.js]: i realize we're using the term scale in like 3 different ways here... 😬 sorry

Comment (48klocs) [reply src/app/destinyTrackerApi/d2-reviewDataCache.js]: As we slice things more finely (platforms, game mode, sandbox rebalances), we'll be dealing with smaller sets of ratings, so I'm not expecting to have huge numbers of ratings for everything.  Rather than use these hard magic numbers, we could find the maximum number of votes that a given item has and base the weighting scale off of that.

I'm also OK with things looking curiously bad, because I think that will spur people to throw a thumbs up at it.

Webscale.

Comment (48klocs) [reply src/app/destinyTrackerApi/d2-reviewDataCache.js]: OK, a little new logic is in there - rather than fixed points, it weights based off of total number of votes compared to the max in the user's data.

**Comment by bhollis** [comment]
One worry - won’t this make it so two different users might see a different rating for the same gun?

**Comment by 48klocs** [comment]
Yes, that's a possibility. It's also a possibility right now (if you change your platform, the ratings change).

My thinking is that most people will keep the "best" weapons (with the most votes) and that concern will be more theoretical than anything. For newer people starting out, they're probably less concerned with what the best weapon in the game is and more concerned with what the best weapon that they have in their inventory is, and either of these two versions (weighting relative to votes or relative to max votes for all items in their inventory) can help answer that.

**Comment by kyleshay** [comment]
If it was based on a scale, maybe it could be something that we agree upon with DTR so it is done on their side, that way it's the same rating for everyone?

**Comment by 48klocs** [comment]
DTR is presenting the up and down ratings as just that; we're the ones that are presenting a star rating.

I'd like to keep it a single number in DIM - it's easier to compare and contrast items in a player's inventory that way.

**Comment by kyleshay** [comment]
Ah, ok awesome. Hmm, could we get some metadata back from DTR? Total number of reviews, etc? If not I'm guessing most reviews may look the same as the item w/ the most number of reviews (MIDA?) is owned by the larger population anyway.

**Comment by 48klocs** [comment]
What's in `_setMaximumTotalVotes` could be converted to sum vote count across all items instead of finding the maximum on any one item.

What sort of algorithm were you pondering exploring?

**Comment by kyleshay** [comment]
I just meant if we could get the highest vote count back from DTR then we wouldn't need to calculate `_setMaximumTotalVotes` ourselves and we could use it as a consistent number regardless of what items any user has.

**Comment by kyleshay** [comment]
It's just a minor thing. Other than that I think your algo should work really well!! I'd be fine w/ merging and doing that as another enhancement
